sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_responses.hpp

```diff
@@ -5,63 +5,108 @@
 #include <string>
+#include <thread>
 #include <vector>
 
+#include <azure/core/operation.hpp>
+
 #include "azure/storage/blobs/protocol/blob_rest_client.hpp"
 
-namespace Azure { namespace Storage { namespace Blobs { namespace Models {
-
-  struct DownloadBlobToResult
-  {
-    std::string ETag;
-    Azure::Core::DateTime LastModified;
-    int64_t ContentLength = 0;
-    BlobHttpHeaders HttpHeaders;
-    Storage::Metadata Metadata;
-    Models::BlobType BlobType;
-    bool IsServerEncrypted = false;
-    Azure::Core::Nullable<std::vector<uint8_t>> EncryptionKeySha256;
-  };
-
-  using UploadBlockBlobFromResult = UploadBlockBlobResult;
-
-  struct AcquireBlobLeaseResult
-  {
-    std::string RequestId;
-    std::string ETag;
-    Azure::Core::DateTime LastModified;
-    std::string LeaseId;
-  };
-
-  struct BreakBlobLeaseResult
-  {
-    std::string RequestId;
-    std::string ETag;
-    Azure::Core::DateTime LastModified;
-    int32_t LeaseTime = 0;
-  };
-
-  struct ChangeBlobLeaseResult
-  {
-    std::string RequestId;
-    std::string ETag;
-    Azure::Core::DateTime LastModified;
-    std::string LeaseId;
-  };
-
-  struct ReleaseBlobLeaseResult
-  {
-    std::string RequestId;
-    std::string ETag;
-    Azure::Core::DateTime LastModified;
-  };
-
-  struct RenewBlobLeaseResult
-  {
-    std::string RequestId;
-    std::string ETag;
-    Azure::Core::DateTime LastModified;
-    std::string LeaseId;
-  };
-
-}}}} // namespace Azure::Storage::Blobs::Models
+namespace Azure { namespace Storage { namespace Blobs {
+
+  class BlobClient;
+  class PageBlobClient;
+
+  namespace Models {
+
+    struct DownloadBlobToResult
+    {
+      std::string ETag;
+      Azure::Core::DateTime LastModified;
+      int64_t ContentLength = 0;
+      BlobHttpHeaders HttpHeaders;
+      Storage::Metadata Metadata;
+      Models::BlobType BlobType;
+      bool IsServerEncrypted = false;
+      Azure::Core::Nullable<std::vector<uint8_t>> EncryptionKeySha256;
+    };
+
+    using UploadBlockBlobFromResult = UploadBlockBlobResult;
+
+    struct AcquireBlobLeaseResult
+    {
+      std::string RequestId;
+      std::string ETag;
+      Azure::Core::DateTime LastModified;
+      std::string LeaseId;
+    };
+
+    struct BreakBlobLeaseResult
+    {
+      std::string RequestId;
+      std::string ETag;
+      Azure::Core::DateTime LastModified;
+      int32_t LeaseTime = 0;
+    };
+
+    struct ChangeBlobLeaseResult
+    {
+      std::string RequestId;
+      std::string ETag;
+      Azure::Core::DateTime LastModified;
+      std::string LeaseId;
+    };
+
+    struct ReleaseBlobLeaseResult
+    {
+      std::string RequestId;
+      std::string ETag;
+      Azure::Core::DateTime LastModified;
+    };
+
+    struct RenewBlobLeaseResult
+    {
+      std::string RequestId;
+      std::string ETag;
+      Azure::Core::DateTime LastModified;
+      std::string LeaseId;
+    };
+
+    struct StartCopyBlobResult : public Azure::Core::Operation<GetBlobPropertiesResult>
```

Comment:
Should be class instead of struct?

Reply:
I don't think there is any essential difference between `struct ` and `class`. But other return types are `struct`, so

Reply:
https://github.com/Azure/azure-sdk/issues/1974
Topic: Struct vs Class
When should code be in a struct vs in a class?
Conclusion:
•	Structs are for passive objects that carry data;  everything else is a class
o	Invariants between fields are disallowed
o	Constructors/Destructors are allowed on structs but must not enforce any invariant
•	When in doubt make it a class.

We did mention that for non-passive objects, it should not be structs.